### ISSUE TITLE
SC1044 breaks on empty quoted strings in heredoc delimeter word 

### ISSUE THREAD

**Issue by jrogov** [koalaman/shellcheck#2726]
#### For bugs
- Rule Ids: 
  - SC1044 (main issue)
  - SC1073 and SC1009 (transitive)
  - SC1072 (as a consequence)
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
```
ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

Quoted version
```bash
#!/usr/bin/env bash

set a b c

cat <<EOF
$1 $2 $3
EOF

cat <<'EOF'
$1 $2 $3
EOF

cat <<''EOF
$1 $2 $3
EOF
```

or 

Doublequoted version:
```bash
#!/usr/bin/env bash

set a b c

cat <<EOF
$1 $2 $3
EOF

cat <<"EOF"
$1 $2 $3
EOF

cat <<""EOF
$1 $2 $3
EOF
```

#### Execution result (output)
```
a b c
$1 $2 $3
$1 $2 $3
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 13:
cat <<""EOF
^-- SC1009 (info): The mentioned syntax error was in this simple command.
    ^-- SC1073 (error): Couldn't parse this here document. Fix to allow more checks.
      ^-- SC1044 (error): Couldn't find end token `""EOF' in the here document.
 
Line 16:

^-- SC1072 (error): Here document was not correctly terminated. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:
No errors at all

### Notes

I see there exists [a closed issue very similar issue](https://github.com/koalaman/shellcheck/issues/1650), but in reality [PR mentioned](https://github.com/koalaman/shellcheck/pull/1735) there haven't fixed the specific usecase above (using empty quoted string to disable expansion in heredoc).

Here's the exceprt from bash man just in case:
> No  parameter  and variable expansion, command substitution, arithmetic
       expansion, or pathname expansion is performed on word.  I**f any part  of
       word  is  quoted, the delimiter is the result of quote removal on word,
       and the lines in the  here-document  are  not  expanded.**   If  word  is
       unquoted,  all  lines  of  the here-document are subjected to parameter
       expansion, command substitution, and arithmetic expansion, the  charac-
       ter  sequence  \<newline>  is  ignored, and \ must be used to quote the
       characters \, $, and `.

**Comment by Drew-Daniels** [comment]
Same issue here
